graph-gateway/src/main.rs

```diff
@@ -98,9 +103,27 @@ async fn main() {
             conf.attestations.dispute_manager,
         )));
 
+    let legacy_signer: &'static SecretKey = Box::leak(Box::new(
+        conf.scalar
+            .legacy_signer
+            .map(|s| s.0)
+            .unwrap_or(conf.scalar.signer.0),
+    ));
+    let receipt_signer: &'static ReceiptSigner = Box::leak(Box::new(ReceiptSigner::new(
+        conf.scalar.signer.0,
+        conf.scalar.chain_id,
+        conf.scalar.verifier,
+        legacy_signer,
+    )));
+
     // Initialize the network service and wait for the initial network state synchronization
-    let network_subgraph_client =
-        SubgraphClient::new(http_client.clone(), conf.network_subgraph.clone());
+    let network_subgraph_client = init_subgraph_client(
+        http_client.clone(),
+        receipt_signer,
+        0, // TODO <------------------------------------ Pay the MAX per query
```

Comment:
I need help with this:

> How can I get at this point the maximum amount of GRT to pay an indexer per query?

Reply:
Since the cost of there queries doesn't impact consumer balances, we don't need to involve the budgeting system. We can simply take the gateway's per-query budget directly, and pay the indexers that fixed price. In the config, it's `query_fees_target`.

Reply:
We could also pay indexers the minimum value we ever send, 10e-6. But I think these queries are sufficiently complex & infrequent that they deserve the max price.